src/lib/components/DashNgl.react.js

```diff
@@ -101,36 +77,28 @@ export default class DashNgl extends Component {
 
             // translate by x angstrom along chosen axis
             stageObj.setPosition([xOffset, 0, 0]);
-            // stage.animationControls.rotate(pa.getRotationQuaternion(),1500)
         } else {
             stageObj.addRepresentation('cartoon');
         }
         stage.animationControls.moveComponent(stageObj, stageObj.getCenter());
-        // stage.autoView()
     }
 
-    // If user has selected structure already just add the new Representation
+    // If user has selected structure load it from the browser
     loadStructure(stage, filename, chain, color, xOffset) {
-        console.log('load from browser');
-        // console.log(filename)
         const stageObj = stage.getComponentsByName(filename).list[0];
```

Comment:
filename is literally the filename how it is named in the data folder (e.g. 6CHG.pdb). It is taken from the information coming from the backend to the frontend. These are collected in `componentDidUpdate`:
`const {data, stageParameters} = this.props;`

And the filename is attached in the loadData method to the stageObj see code snippet below
```
loadFile(stringBlob, {ext: data.ext, defaultRepresentation: false}).then(stageObj => {
                stageObj.name = data.filename;
```
This naming of the stageObj. is then the basis for the method `loadStructure` with its `getComponentsByName` function

Maybe this snippet makes the whole process more clear:
https://github.com/arose/ngl/blob/e8713514cfa7100a8313b3ff34a3c94b4125ced4/examples/scripts/component/getByName.js

The idea behind all this as soon as someone picks a molecule it is loaded and saved in the stageObj.  which isn't cleared as long the user does not reload the page. So if the user selects some different molecules but from time to time comes back to its very first selection. it's structure is still loaded in the application so there is no need for another backend call.

This is really important for our specific use case since we want the webserver calls as low as possible and so far it did not lead to too high ram usage of the browser. However I am not sure if that approach would be suitable for mobile users as well.

Up to now we did not any stress tests yet. Meaning how much structures I can load into the stage till this system breaks would be interesting to get to know the limitations. :dagger: